src/transactions.ts

```diff
@@ -35,19 +34,28 @@ export class TransactionOnNetwork {
         Object.assign(this, init);
     }
 
-    static fromProxyHttpResponse(txHash: string, response: any): TransactionOnNetwork {
+    static fromProxyHttpResponse(txHash: string, response: any, processStatus?: TransactionStatus | undefined): TransactionOnNetwork {
         let result = TransactionOnNetwork.fromHttpResponse(txHash, response);
         result.contractResults = ContractResults.fromProxyHttpResponse(response.smartContractResults || []);
-        result.isCompleted = new TransactionCompletionStrategyOnProxy().isCompleted(result);
-        // TODO: uniformize transaction status.
+
+        if (processStatus !== undefined) {
+            result.status = processStatus;
+
+            if (result.status.isSuccessful() || result.status.isFailed()) {
```

Comment:
This code 
```
if (result.status.isSuccessful() || result.status.isFailed()) {
  result.isCompleted = true;
}
else {
  result.isCompleted = false;
}
```
could be reduced to:

`result.isCompleted = result.status.isSuccessful() || result.status.isFailed()`

Reply:
Done! Thanks!